### PR TITLE
Feat/workspaces

### PR DESCRIPTION
**Pull request by cheikhgwane**
A short, meaningful PR description. Explain the goal of the PR and the ideas behind it.

## Changes

Please list / describe the changes in the codebase for the reviewer(s).

- Change 1
- Change 2

## How/what to test

A few words about what needs to be tested, along with specific testing instructions if needed.

## Screenshots / screencast

If relevant, please add some screenshots or a short video.

### CODE REVIEW

**Comment by qgerome** [comment]
CleanShot 2022-11-24 at 14 20 55@2x
CleanShot 2022-11-24 at 14 18 44@2x

**Comment by qgerome** [comment]
I think it's because of me but could you fix the tests and rebase main please ?

**Comment by qgerome** [comment]
CleanShot 2022-11-24 at 14 58 36@2x
The iframe does not take the full height. You need to add the header of the page and maybe remove the padding around the iframe as well.

**Comment by cheikhgwane** [comment]
> CleanShot 2022-11-24 at 14 20 55@2x CleanShot 2022-11-24 at 14 18 44@2x

Okay, will update it.

**Comment by cheikhgwane** [comment]
> I think it's because of me but could you fix the tests and rebase main please ?

Copy that.

**Review by qgerome — CHANGES_REQUESTED** [review]

File: src/pages/workspaces/[workspaceId]/connections/[connectionId].tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+                <CodeEditor
+                  readonly
+                  lang="json"
+                  value={workspace.database.workspaceTables[0].codeSample[0]}
+                />
+              </div>
+            </Tabs.Tab>
+            <Tabs.Tab className="mt-4" label={t("Credentials")}>
+              <div className="mt-5">
+                <p>Creds 1 : Value 1</p>
```
[the remark this answers is not in the document]
Comment (cheikhgwane) [reply src/pages/workspaces/[workspaceId]/connections/[connectionId].tsx:96]: Copy that

File: src/pages/notebooks/index.tsx [hunk 2]
```diff
@@ -11,11 +11,7 @@ const NotebooksPage = () => {
     return null;
   }
 
-  return (
```
Comment (qgerome) [inline src/pages/notebooks/index.tsx:14]: Why do you edit this page ?

**Review by qgerome — CHANGES_REQUESTED** [review]

File: src/pages/notebooks/index.tsx (hunk 2 above)
Comment (cheikhgwane) [reply src/pages/notebooks/index.tsx:14]: Ooh that's a mistake. Let me get back the content

Comment (cheikhgwane) [reply src/pages/notebooks/index.tsx:14]: Ooh that's a mistake. Let me get back the content

**Comment by qgerome** [comment]
CleanShot 2022-11-28 at 10 55 20@2x
Hostname, username, password, port are _Fields_ that have to be added to the form at the bottom.

As you can see buttons are not displayed when you select postgresql. Please add a title to the dialog and use the standard way of adding button to a dialog.

**Comment by qgerome** [comment]
Layout of the connection card is buggy:
CleanShot 2022-11-28 at 11 02 34@2x

**Comment by qgerome** [comment]
> Layout of the connection card is buggy:
> CleanShot 2022-11-28 at 11 02 34@2x

It would make sense to be able to go to the connection by clicking on its title.

**Comment by cheikhgwane** [comment]
> > Layout of the connection card is buggy:
> > CleanShot 2022-11-28 at 11 02 34@2x
> 
> It would make sense to be able to go to the connection by clicking on its title.

Okay. So for the settings and usage&details link what pages should they redirect? Or i can leave the href blank for now?

**Review by qgerome — CHANGES_REQUESTED** [review]

File: src/pages/workspaces/[workspaceId]/connections/index.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+          </Breadcrumbs.Part>
+          <Breadcrumbs.Part
+            href={`/workspaces/${encodeURIComponent(workspace.id)}/Connections`}
+          >
+            {t("Connections")}
+          </Breadcrumbs.Part>
+        </Breadcrumbs>
+      </WorkspaceLayout.Header>
+      <WorkspaceLayout.PageContent>
+        <div className="grid grid-cols-3 gap-5 sm:grid-cols-3">
```
[the remark this answers is not in the document]
Comment (cheikhgwane) [reply src/pages/workspaces/[workspaceId]/connections/index.tsx:47]: Okay

File: src/pages/workspaces/[workspaceId]/connections/index.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+          <div className="col-start-1 col-end-4 flex justify-end">
+            <div>
+              <Button onClick={() => setOpenModal(true)}>
```
[the remark this answers is not in the document]
Comment (cheikhgwane) [reply src/pages/workspaces/[workspaceId]/connections/index.tsx:50]: Yes. Will add it

File: src/workspace/features/ConnectionDataCard/ConnectionDataCard.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+              query: { connectionId: connection.id },
+            }}
+          >
+            <Cog8ToothIcon className="h-6 w-6" />
+            {t("Settings")}
+          </Link>
+          <Link
+            className="flex items-end space-x-2 text-blue-500 text-blue-500"
+            href={{
+              pathname: `${asPath}/[connectionId]`,
```
[the remark this answers is not in the document]
Comment (cheikhgwane) [reply src/workspace/features/ConnectionDataCard/ConnectionDataCard.tsx]: Got it.

File: src/pages/workspaces/[workspaceId]/databases/[tableId].tsx [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+      <WorkspaceLayout.Header>
+        <Breadcrumbs withHome={false}>
+          <Breadcrumbs.Part
+            isFirst
+            href={`/workspaces/${encodeURIComponent(workspace.id)}`}
+          >
+            {workspace.name}
+          </Breadcrumbs.Part>
+          <Breadcrumbs.Part
+            href={`/workspaces/${encodeURIComponent(workspace.id)}/Database`}
```
Comment (qgerome) [inline src/pages/workspaces/[workspaceId]/databases/[tableId].tsx:56]: Better to keep the urls in lower case.

**Review by qgerome — CHANGES_REQUESTED** [review]

File: src/pages/workspaces/[workspaceId]/databases/[tableId].tsx (hunk 6 above)
Comment (cheikhgwane) [reply src/pages/workspaces/[workspaceId]/databases/[tableId].tsx]: Updated

File: src/pages/workspaces/[workspaceId]/databases/[tableId].tsx [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+      <WorkspaceLayout.Header>
+        <Breadcrumbs withHome={false}>
+          <Breadcrumbs.Part
+            isFirst
+            href={`/workspaces/${encodeURIComponent(workspace.id)}`}
+          >
+            {workspace.name}
+          </Breadcrumbs.Part>
+          <Breadcrumbs.Part
+            href={`/workspaces/${encodeURIComponent(workspace.id)}/database`}
```
Comment (qgerome) [inline src/pages/workspaces/[workspaceId]/databases/[tableId].tsx:56]: singular

File: src/pages/workspaces/[workspaceId]/databases/[tableId].tsx [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+          >
+            {t("Database")}
+          </Breadcrumbs.Part>
+          <Breadcrumbs.Part
+            href={`/workspaces/${encodeURIComponent(workspace.id)}/databases/${
```
Comment (qgerome) [inline src/pages/workspaces/[workspaceId]/databases/[tableId].tsx:61]: plural

=> Use the singular

**Review by qgerome — CHANGES_REQUESTED** [review]

File: src/pages/workspaces/[workspaceId]/databases/[tableId].tsx (hunk 7 above)
Comment (cheikhgwane) [reply src/pages/workspaces/[workspaceId]/databases/[tableId].tsx:56]: Yes, It should be /databases

**Pull request merged**